### PR TITLE
RFD 0130: Autogenerate the Teleport Resource Reference

### PR DESCRIPTION
**Pull request by ptgott**
Add an RFD to automatically generate the Teleport resource reference. While the documentation lists the resources that are available to apply via `tctl`, this list is incomplete, and it has been difficult to keep the docs up to date with changes to Teleport resources. As a result, this RFD proposes generating the resource reference from Go source files.

### CODE REVIEW

**Comment by ptgott** [comment]
See the discussion in the original PR: https://github.com/gravitational/teleport/pull/25823

**Review by codingllama — APPROVED** [review]
LGTM, assuming no changes in relation to #25823.

Please add a brief PR description.

File: rfd/0128-autogenerate-resource-reference.md [hunk 1]
```diff
@@ -0,0 +1,600 @@
...
+configuration objects that Teleport users can apply using `tctl`, Terraform, and
+other methods.
+
+The scope of this RFD is to automatically generate the "Dynamic resources"
+section of the resource reference (`docs/pages/reference/resources.mdx`), a list
+of the dynamic resources you can apply via `tctl`, from the Teleport source
+code. 
+
+To do this, write a program that we can run as a new Make target in a clone of
+`gravitational/teleport`. When we make changes to Teleport's dynamic resources ,
```
Comment (zmb3) [inline rfd/0128-autogenerate-resource-reference.md:24]:
```suggestion
`gravitational/teleport`. When we make changes to Teleport's dynamic resources,
```

File: rfd/0128-autogenerate-resource-reference.md [hunk 2]
```diff
@@ -0,0 +1,600 @@
...
+For resources that we want to _exclude_ from the reference, the generator will
+read from a mapping of package paths to lists of named struct types within each
+package. 
+
+The program will declare the mapping within the Go code as a `map`.
+
+Here is an example:
+
+```go
+var resources = map[string][]string{
```
Comment (zmb3) [inline rfd/0128-autogenerate-resource-reference.md:73]:
```suggestion
var exclude = map[string][]string{
```

**Review by zmb3 — APPROVED** [review]

File: rfd/0128-autogenerate-resource-reference.md [hunk 3]
```diff
@@ -0,0 +1,600 @@
...
+|Field Name|Description|Type|
+|---|---|---|
+|`uri`|The web app endpoint.|`string`|
+|`public_addr`|The public address the application is accessible at.|`string`|
+|`dynamic_labels`|The app's command labels.|`map[string]`[Command LabelV2](#command-label-v2)|
+|`insecure_skip_verify`|Disables app's TLS certificate verification.|`boolean`|
+|`rewrite`|A list of rewriting rules to apply to requests and responses.|[Rewrite](#rewrite)|
+|`aws`|Contains additional options for AWS applications.|[App AWS](#app-aws)|
+|`cloud`|Identifies the cloud instance the app represents.|`string`|
+
```
Comment (alexfornuto) [inline rfd/0128-autogenerate-resource-reference.md:184]: Should the template create a header or other copy to specify that this is an excerpt from a specific config file and/or resource definition?

**Review by alexfornuto — APPROVED** [review]
One comment (previous review), generally 👍

File: rfd/0128-autogenerate-resource-reference.md (hunk 3 above)
Comment (ptgott) [reply rfd/0128-autogenerate-resource-reference.md:184]: Do you mean that, if a type is a field within a resource definition, the reference for the field type should list the resource definitions it's part of?

Comment (ptgott) [reply rfd/0128-autogenerate-resource-reference.md:184]: I'm pretty sure this comment refers to something the template includes already—the source of each `tctl`-applicable resource—so I'm going to go ahead and merge this. Feel free to open a PR/issue if there's anything I should change!

**Pull request merged**